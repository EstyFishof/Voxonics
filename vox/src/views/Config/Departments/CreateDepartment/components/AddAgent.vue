<template>
  <div class="personal-info-wrap">
    <div class="wrap-head">
      <h2>ADD AGENTS</h2>
      <span class="hr"></span>
    </div>
    <div class="top">
      <div class="left">Add agent
      </div>
      <div class="agents users">
        <div v-if="newDepartmentAgents.length" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentAgents.length > 1" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentAgents.length > 2" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="newDepartmentAgents.length > 3" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
      </div>
    </div>
    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <div class="label" v-click-outside="hideListUsers">
              <input placeholder="Enter name or Agent Number"
                     class="input-field"
                     @click="showList()"
                     @input="showAgents = true; searchDepartments()"
                     v-model.trim="name"/>
              <span @click="showList()" class="triangle" :class="{'rotate-arrow': showAgents}">
                <img src="@/assets/img/icons/select-arrow.svg" alt="select">
              </span>
              <DropdownList v-show="showAgents" :classNameList="'list'" :dataList="mainList.getMainList"  @scrollFunc="(event) => funScroll(event)"
                  @clickFunc="(item) => chooseAgent(item)" :itemText="(item) => `<span>${item.name}</span><span>ID ${item.internalNumber}</span>`"
                  :classNameItemComputed="(item) => newDepartmentAgents.filter(element => element.uid === item.uid).length ? collapse ? 'selected collapsed row' : 'selected row' : 'row'"
                  :showItem="(item) => newDepartmentAgents.filter(element => element.uid === item.uid).length ? !collapse : true">
                <span class="collapse" @click="collapse = !collapse">
                  <img class="button" :src="collapse ? extendImg : collapseImg" alt="collapse">
                  <span class="text">Selected users available to you:</span>
                  <span class="amount">{{ mainList.topList.length }}</span>
                </span>
              </DropdownList>
            </div>
          </div>
        </div>
      </div>
      <div class="agents users">
        <div class="agent user" v-for="(agent, index) of newDepartmentAgents" :key="index">
          <div class="block">
            <div class="round"><img :src="agent.avatar || userAva" alt="avatar"></div>
            <span class="name">{{ agent.name }}</span>
          </div>
          <span class="name">{{ agent.internalNumber }}</span>
          <img @click="chooseAgent(agent)" class="union" src="@/assets/img/icons/union.svg" width="19" alt="">
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import userAva from '@/assets/img/icons/user_icon.svg'
import collapseImg from '@/assets/img/icons/collapse.svg'
import extendImg from '@/assets/img/icons/extend.svg'
import DropdownList from '@/components/DropdownList'
import ClickOutside from "vue-click-outside"
import {dropdownListFetch, dropdownChooseItem} from '@/lib/dropdownListsDepsAndQueues.js'
import _ from 'lodash'

export default {
  name: "AddAgent",
  components: {
    DropdownList
  },
  data() {
    return {
      collapseImg,
      extendImg,
      collapse: false,
      name: "",
      showAgents: false,
      userAva: userAva,
      perPage: 9,
      mainList: {
        topList: [],
        bottomList: [],
        getMainList: []
      },
    }
  },
  watch: {
    getGlobalUsersList() {
      dropdownListFetch(this, this.getGlobalUsersList, 'newDepartmentAgents')
      this.mainList.getMainList = _.concat(this.mainList.topList, this.mainList.bottomList)
    }
  },
  computed: {
    ...mapGetters(['newDepartmentAgents', 'getGlobalUsersList', 'getPaginatorUsers']),
  },
  methods: {
    searchDepartments() {
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch(), 800)
    },
    funScroll({target}) {
      if(((target.clientHeight + target.scrollTop === target.scrollHeight) && (this.getPaginatorUsers.paginator.count > this.perPage))) {
        this.perPage += 9
        this.fetch()
      }
    },
    fetch() {
      this.$store.dispatch('globalUsersListRequest', {peerPage: this.perPage, sort: '', page: 1, search: this.name})
    },
    showList() {
      if(!this.showAgents) {
        this.fetch()
      }
      this.showAgents = !this.showAgents
    },
    hideListUsers() {
      this.showAgents = false
    },
    chooseAgent(agent) {
      dropdownChooseItem(this, agent, 'newDepartmentAgents', 'setAgentsDepartment')
    },
    setAgentsDepartment(array) {
      this.$store.dispatch('transferAgentsDepartment', array)
    }
  },
  mounted() {
    this.popupItem = this.$el.children[2].children[0]
    if (this.newDepartmentAgents.length) {
      this.mainList.topList = this.newDepartmentAgents
    }
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/departmentsAndQueues/agentsSettings.scss"></style>

<style scoped lang="scss">
.personal-info-wrap {
  background: none;
  padding-left: 0;
  border-top: none;
  height: 363px;

  .wrap-head {
    h2 {
      font-size: 48px;
      margin-right: 24px;
    }
  }
  
  .top {
    margin: 50px 24px 0 24px;

    .left {
      margin-right: 44px;
    }
  }

  .wrap-body {
    margin: 0 0 36px 0;

    .personal-info {

      &__first-part {
        .item-container {
          margin: 0 24px 16px 24px;

          .label {
            .list {
              height: 175px;
            }
          } 
        }
      }
    }
  }

  .agents {
    .agent {
      color: var(--font-panel);
    }
  }
}
@media (max-width: 1780px) {
  .personal-info-wrap {
    .top {
      margin: 50px 0 0 0;

      .left {
        margin-right: 20px;
        margin-left: 24px;
      }
    }
    .wrap-body {
      .personal-info__first-part {
        .item-container {
          margin: 0 0 16px 24px;
        }
      }
    }
  }
}
</style>
