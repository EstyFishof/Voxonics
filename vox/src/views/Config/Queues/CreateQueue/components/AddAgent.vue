<template>
  <div class="personal-info-wrap">
    <div class="wrap-head">
      <h2>ADD AGENTS</h2>
      <span class="hr"></span>
    </div>
    <div class="top">
      <div class="left">Add agent
      </div>
      <div class="agents users">
        <div v-if="relatedAgents.length" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="relatedAgents.length > 1" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="relatedAgents.length > 2" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
        <div v-if="relatedAgents.length > 3" class="agent user">
          <div class="block">
            <div class="round">Name</div>
            <span class="name"></span>
          </div>
          <span class="name">Agent Number</span>
          <span class="union"></span>
        </div>
      </div>
    </div>
    <div class="wrap-body">
      <div class="personal-info">
        <div class="personal-info__first-part">
          <div class="item-container with-err">
            <div class="label" v-click-outside="hideListUsers">
              <input placeholder="Enter name or Agent Number"
                     class="input-field"
                     @click="showList()"
                     @input="showAgents = true; searchQueues()"
                     v-model.trim="name"/>
              <span @click="showList()" class="triangle" :class="{'rotate-arrow': showAgents}">
                <img src="@/assets/img/icons/select-arrow.svg" alt="select">
              </span>

              <DropdownList v-show="showAgents" :classNameList="'list'" :dataList="mainList.getMainList"  @scrollFunc="(event) => funScroll(event)"
                  @clickFunc="(item) => chooseAgent(item)" :itemText="(item) => `<span>${item.name}</span><span>ID ${item.internalNumber}</span>`"
                  :classNameItemComputed="(item) =>  relatedAgents.filter(element => element.uid === item.uid).length ? collapse ? 'selected collapsed row' : 'selected row' : 'row'"
                  :showItem="(item) => relatedAgents.filter(element => element.uid === item.uid).length ? !collapse : true">
                <span class="collapse" @click="collapse = !collapse">
                  <img class="button" :src="collapse ? extendImg : collapseImg" alt="collapse">
                  <span class="text">Selected users available to you:</span>
                  <span class="amount">{{ mainList.topList.length }}</span>
                </span>
              </DropdownList>
            </div>
          </div>
        </div>
      </div>
      <div class="agents users">
        <drop-list :items="relatedAgents" @reorder="$event.apply(relatedAgents); reorderAgents()" @insert="insert1" mode="cut">
          <template v-slot:item="{item}">
            <drag :key="item.uid" :data="item" @cut="remove(relatedAgents, item)">
              <div class="agent user">
                <div class="block">
                  <div class="number">{{relatedAgents.indexOf(item) + 1}}.</div>
                  <div class="round"><img :src="item.avatar || userAva" alt="avatar"></div>
                  <span class="name">{{ item.name }}</span>
                </div>
                <span class="name">{{ item.internalNumber }}</span>
                <img @click="chooseAgent(item)" class="union" src="@/assets/img/icons/union.svg" width="19" alt="">
              </div>
            </drag>
          </template>
        </drop-list>
      </div>
    </div>
  </div>
</template>

<script>
import { mapGetters } from "vuex"
import userAva from "@/assets/img/icons/user_icon.svg"
import collapseImg from '@/assets/img/icons/collapse.svg'
import extendImg from '@/assets/img/icons/extend.svg'
import ClickOutside from "vue-click-outside"
import DropdownList from '@/components/DropdownList'
import { Drag, DropList } from "vue-easy-dnd"
import {dropdownListFetch, dropdownChooseItem} from '@/lib/dropdownListsDepsAndQueues.js'
import _ from 'lodash'

export default {
  name: "AddAgent",
  components: {Drag, DropList, DropdownList},
  data() {
    return {
      collapseImg,
      extendImg,
      collapse: false,
      name: "",
      showAgents: false,
      relatedAgents: [],
      userAva: userAva,
      perPage: 9,
      mainList: {
        topList: [],
        bottomList: [],
        getMainList: []
      },
    }
  },
  watch: {
    getGlobalUsersList() {
      dropdownListFetch(this, this.getGlobalUsersList, 'relatedAgents')
      this.mainList.getMainList = _.concat(this.mainList.topList, this.mainList.bottomList)
    }
  },
  computed: {
    ...mapGetters(['newQueueAgents', 'userInfo', 'getGlobalUsersList', 'getPaginatorUsers']),
  },
  methods: {
    searchQueues() {
      clearTimeout(this.timer)
      this.timer = setTimeout(() => this.fetch(), 800)
    },
    funScroll({target}) {
      if ((target.clientHeight + target.scrollTop === target.scrollHeight) && (this.getPaginatorUsers.paginator.count > this.perPage)) {
        this.perPage += 9
        this.fetch()
      }
    },
    fetch() {
      this.$store.dispatch('globalUsersListRequest', {peerPage: this.perPage, sort: '', page: 1, search: this.name, role: '', order: 'asc'})
    },
    hideListUsers() {
      this.showAgents = false
    },
    showList() {
      if (!this.showAgents) {
        this.fetch()
      }
      this.showAgents = !this.showAgents
    },
    chooseAgent(agent) {
      dropdownChooseItem(this, agent, 'relatedAgents', 'setAgentsQueue', true)
    },
    setAgentsQueue(array) {
      this.$store.dispatch('setRelatedWithQueue', array)
    },
    reorderAgents() {
      this.setAgentsQueue(this.relatedAgents)
    },
    insert1(event) {
      let array = this.relatedAgents.splice(event.index, 0, event.data)
      this.setAgentsQueue(array)
    },
    remove(array, value) {
      let index = array.indexOf(value)
      array.splice(index, 1)
      this.setAgentsQueue(array)
    },
  },
  mounted() {
    this.popupItem = this.$el.children[2].children[0]
    if (this.newQueueAgents.length) {
      this.relatedAgents = this.newQueueAgents
      this.mainList.topList = this.newQueueAgents
    }
  },
  directives: {
    ClickOutside
  }
}
</script>

<style scoped lang="scss" src="@/assets/styles/departmentsAndQueues/agentsSettings.scss"></style>
<style scoped lang="scss">
.personal-info-wrap {
  background: none;
  padding-left: 0;
  border-top: none;
  height: 363px;
  .drop-list {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
  }

  .wrap-head {
    h2 {
      font-size: 48px;
      margin-right: 24px;
    }
  }

  .top {
    margin: 50px 24px 0 24px;

    .left {
      margin-right: 44px;
    }
  }

  .wrap-body {
    margin: 0 0 36px 0;

    .personal-info {

      &__first-part {
        .item-container {
          margin: 0 24px 16px 24px;

          .label {
            .list {
              height: 175px;
            }
          }
        }
      }
    }
  }

  .agents {
    .agent {
      color: var(--font-panel);
    }
  }
}
@media (max-width: 1780px) {
  .personal-info-wrap {
    .top {
      margin: 50px 0 0 0;
      .left {
        margin-right: 20px;
        margin-left: 24px;
      }
    }
    .wrap-body {
      .personal-info__first-part {
        .item-container {
          margin: 0 0 16px 24px;
        }
      }
    }
  }
}
</style>
