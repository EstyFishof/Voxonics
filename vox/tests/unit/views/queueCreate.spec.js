import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import Queue from '../../../src/views/Config/Queues/CreateQueue/CreateQueue.vue';
import AddAgent from '../../../src/views/Config/Queues/CreateQueue/components/AddAgent.vue';
import AddManagers from '../../../src/views/Config/Queues/CreateQueue/components/AddManagers.vue';
import QueueInfo from '../../../src/views/Config/Queues/CreateQueue/components/QueueInfo.vue';
import VueTippy, { TippyComponent } from "vue-tippy";


const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(VueTippy);
localVue.component("tippy", TippyComponent);
const router = new VueRouter();
localVue.use(Vuex);

describe('CreateQueue', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

const store = new Vuex.Store({
    state: {
        queue: {uid: 5000, strategy: 'Random', option: {edit: false}, ringinuse: 'ringinuse', relatedWith: [], belongTo: []}
    },
    // effects: {
    //     setAgents(state, value) {
    //         state.queue = value
    //     }
    // },
    getters: {
        newQueue(state) {
            return state.queue;
        }
    }
});
    const wrapper = shallowMount(Queue, { store, localVue, stubs: {Queue: true} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('handlerSubmit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'handlerSubmit');
        await wrapper.vm.handlerSubmit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('mounted component', () => {
        Queue.mounted.call(wrapper.vm);
    });
});

describe('AddAgent', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            agentsQueue: [],
            agents: []
        },
        actions: {
            setRelatedWithQueue({commit}, value) {},
        },
        mutations: {
            setQueueAgents(state, value) {
                state.agentsQueue = value;
            }
        },
        getters: {
            newQueueAgents(state) {
                return state.agentsQueue;
            },
            agentsQueue(state) {
                return state.agents;
            }
        }
    });
    const wrapper = shallowMount(AddAgent, { store, localVue, stubs: {AddAgent: true} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('sort typeof', () => {
        expect(typeof wrapper.vm.name).toBe('string');
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await wrapper.vm.chooseAgent({});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await wrapper.setData({relatedAgents: [{uid: 111}]});
        await wrapper.vm.chooseAgent({uid: 111});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({name: 'aaa'})
    //     AddAgent.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({name: '123'})
    //     AddAgent.computed.agentsFiltered.call(wrapper.vm)
    // })
    it('mounted component', async () => {
        const wrapper = shallowMount(AddAgent, { store, localVue, router });
        await store.commit('setQueueAgents', [{}, {}]);
        await AddAgent.mounted.call(wrapper.vm);
    });
});

describe('AddManagers', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            managersQueue: [{option: {edit: true}}],
            managers: [],
            userInfo: {
                option: {
                    edit: false, 
                    internalNumber: 136,
                },
                permission: {
                    role: 'maganer'
                },
                internalNumber: 124
            },
            newQueueManagers: [
                {
                    avatar: '',
                    name: '',
                    internalNumber: '',
                    option: {
                        edit: false
                    }
                }
            ]
        },
        actions: {
            setBelongToQueue({commit}, value) {},
        },
        mutations: {
            setQueueManagers(state, value) {
                state.managersQueue = value;
            }
        },
        getters: {
            newQueueManagers(state) {
                return state.managersQueue;
            },
            managersQueue(state) {
                return state.managers;
            },
            userInfo(state) {
                return state.userInfo;
            }
        }
    });
    const wrapper = shallowMount(AddManagers, { store, localVue, stubs: {AddManagers: true} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('sort typeof', () => {
        expect(typeof wrapper.vm.name).toBe('string');
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await store.commit('setQueueManagers', []);
        await wrapper.vm.chooseAgent({});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('chooseAgent toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'chooseAgent');
        await wrapper.setData({relatedManagers: [{uid: 111}]});
        await wrapper.vm.chooseAgent({uid: 111});
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('changeEdit toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'changeEdit');
        await wrapper.vm.changeEdit();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({name: 'aaa'})
    //     AddManagers.computed.agentsFiltered.call(wrapper.vm)
    // })
    // it('computed agentsFiltered', () => {
    //     wrapper.setData({name: '123'})
    //     AddManagers.computed.agentsFiltered.call(wrapper.vm)
    // })
    it('mounted component', async () => {
        const wrapper = shallowMount(AddManagers, { store, localVue, router });
        await wrapper.setData({relatedManagers: [{option: {edit: true}}]});
        await store.commit('setQueueManagers', [{}, {}]);
        await AddManagers.mounted.call(wrapper.vm);
    });
});

describe('QueueInfo', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    const store = new Vuex.Store({
        state: {
            queue: {},
            queueName: 'name',
            queueNumber: 1111,
            strategy: 'Ring all',
            timeout: 10,
            ring: 'yes'
        },
        actions: {
            setNameQueue({commit}, value) {},
            setInternalNumberQueue({commit}, value) {},
            setStrategyQueue({commit}, value) {},
            setTimeoutQueue({commit}, value) {},
            setRingQueue({commit}, value) {},
        },
        mutations: {
            nameMutations(state, value) {
                state.queueName = value;
            },
            numberMutations(state, value) {
                state.queueNumber = value;
            },
            strategyMutations(state, value) {
                state.strategy = value;
            },
            timeoitMutations(state, value) {
                state.timeout = value;
            },
            ringMutations(state, value) {
                state.ring = value;
            },
        },
        getters: {
            newQueue(state) {
                return state.queue;
            },
            newQueueName(state) {
                return state.queueName;
            },
            newQueueInternalNumber(state) {
                return state.queueNumber;
            },
            newQueueStrategy(state) {
                return state.strategy;
            },
            newQueueTimeout(state) {
                return state.timeout;
            },
            newQueueRing(state) {
                return state.ring;
            },
        }
    });
    const wrapper = shallowMount(QueueInfo, { store, localVue, router, stubs: {QueueInfo: true}, propsData: {requiredFields: {name: true, internalNumber: false}} });

    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('setQueueName toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setQueueName');
        await wrapper.vm.setQueueName();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setQueueNumber toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setQueueNumber');
        await wrapper.vm.setQueueNumber();
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('setShowOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'setShowOptions');
        await wrapper.vm.setShowOptions('strategy');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'selectOptions');
        await wrapper.vm.selectOptions('strategy', 'ring All');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'selectOptions');
        await wrapper.vm.selectOptions('timeout', 10);
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'selectOptions');
        await wrapper.vm.selectOptions('ring', 'yes');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('selectOptions toHaveBeenCalled', async () => {
        const spy = jest.spyOn(wrapper.vm, 'selectOptions');
        await wrapper.vm.selectOptions('', 'yes');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
    it('mounted component', async () => {
        const wrapper = shallowMount(QueueInfo, { store, localVue, router, stubs: {QueueInfo: true}, propsData: {requiredFields: {name: true, internalNumber: false}} });
        await store.commit('nameMutations', '');
        await QueueInfo.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        const wrapper = shallowMount(QueueInfo, { store, localVue, router, stubs: {QueueInfo: true}, propsData: {requiredFields: {name: true, internalNumber: false}}});
        await store.commit('numberMutations', '');
        await QueueInfo.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        const wrapper = shallowMount(QueueInfo, { store, localVue, router, stubs: {QueueInfo: true}, propsData: {requiredFields: {name: true, internalNumber: false}} });
        await store.commit('strategyMutations', '');
        await QueueInfo.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        const wrapper = shallowMount(QueueInfo, { store, localVue, router, stubs: {QueueInfo: true}, propsData: {requiredFields: {name: true, internalNumber: false}} });
        await store.commit('timeoitMutations', '');
        await QueueInfo.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        const wrapper = shallowMount(QueueInfo, { store, localVue, router, stubs: {QueueInfo: true}, propsData: {requiredFields: {name: true, internalNumber: false}} });
        await store.commit('ringMutations', '');
        await QueueInfo.mounted.call(wrapper.vm);
    });
    it('mounted component', async () => {
        const wrapper = shallowMount(QueueInfo, { store, localVue, router, stubs: {QueueInfo: true}, propsData: {requiredFields: {name: true, internalNumber: false}} });
        await QueueInfo.mounted.call(wrapper.vm);
    });
});


