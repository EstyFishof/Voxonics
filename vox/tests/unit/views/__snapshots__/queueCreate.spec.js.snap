// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAgent Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      ADD AGENTS
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Add agent
      
    </div>
     
    <div
      class="agents users"
    >
      <!---->
       
      <!---->
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter name or Agent Number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="agents users"
    />
  </div>
</div>
`;

exports[`AddManagers Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      ADD MANAGERS
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="top"
  >
    <div
      class="left"
    >
      Add manager
      
    </div>
     
    <div
      class="managers users"
    >
      <div
        class="manager user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            Name
          </div>
           
          <span
            class="name"
          />
        </div>
         
        <span
          class="name"
        >
          Agent Number
        </span>
         
        <span
          class="union"
        />
      </div>
       
      <!---->
    </div>
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <div
            class="label"
          >
            <input
              class="input-field"
              placeholder="Enter name or Agent Number"
            />
             
            <span
              class="list"
              style="display: none;"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="managers users"
    >
      <div
        class="manager user"
      >
        <div
          class="block"
        >
          <div
            class="round"
          >
            <img
              alt="avatar"
              src=""
            />
          </div>
           
          <span
            class="name"
          >
            
          </span>
        </div>
         
        <span
          class="name"
        >
          
        </span>
         
        <div
          class="action"
        >
          <label
            class="label-container"
          >
            <input
              checked="checked"
              class="view"
              disabled="disabled"
              type="checkbox"
            />
             
            <span
              class="check"
            />
             
            <span
              class="text"
            >
              View
            </span>
          </label>
           
          <label
            class="label-container"
          >
            <input
              class="edit"
              type="checkbox"
            />
             
            <span
              class="check"
            />
             
            <span
              class="text"
            >
              Edit
            </span>
          </label>
        </div>
         
        <img
          alt=""
          class="union"
          src="@/assets/img/icons/union.svg"
          width="19"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`CreateQueue Check snapshot 1`] = `
<div
  class="wrap-main"
>
  <div
    class="wrap-form"
  >
    <form>
      <queueinfo-stub
        requiredfields="[object Object]"
      />
       
      <addagent-stub />
       
      <addmanager-stub />
       
      <div
        class="btn-container"
      >
        <button
          class="btn-submit"
        >
          Apply Changes
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`QueueInfo Check snapshot 1`] = `
<div
  class="personal-info-wrap"
>
  <div
    class="wrap-head"
  >
    <h2>
      QUEUE INFO
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <label>
            
              Name
 
            <input
              class="input-field error"
              maxlength="12"
              placeholder="Enter name of Queue"
              required="required"
            />
          </label>
           
          <label>
            
              Number of Queue
 
            <input
              class="input-field"
              placeholder="Enter number of Queue"
              required="required"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="wrap-head options-title"
  >
    <h2>
      Call Settings
    </h2>
     
    <span
      class="hr"
    />
  </div>
   
  <div
    class="wrap-body"
  >
    <div
      class="personal-info"
    >
      <div
        class="personal-info__first-part"
      >
        <div
          class="item-container with-err"
        >
          <label
            class="strategy-label options-label"
          >
            
              Strategy
 
            <input
              class="input-field num"
              readonly="readonly"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="../../../../../assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                Ring all
              </li>
               
              <li>
                Least recent
              </li>
               
              <li>
                Fewest calls
              </li>
               
              <li>
                Random
              </li>
               
              <li>
                RR memory
              </li>
               
              <li>
                Linear
              </li>
               
              <li>
                W random
              </li>
            </ul>
          </label>
           
          <label
            class="strategy-label options-label"
          >
            
              Timeout
 
            <input
              class="input-field num"
              readonly="readonly"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="../../../../../assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                0
              </li>
               
              <li>
                1
              </li>
               
              <li>
                2
              </li>
               
              <li>
                3
              </li>
               
              <li>
                4
              </li>
               
              <li>
                5
              </li>
            </ul>
          </label>
           
          <label
            class="strategy-label options-label"
          >
            
              Ring in use
 
            <input
              class="input-field num"
              readonly="readonly"
              type="text"
            />
             
            <span
              class="triangle"
            >
              <img
                alt="select"
                src="../../../../../assets/img/icons/queues-select-arrow.png"
              />
            </span>
             
            <ul
              class="list-related-numbers"
              style="display: none;"
            >
              <li>
                Yes
              </li>
               
              <li>
                No
              </li>
            </ul>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
